indexing-hadoop/src/main/java/io/druid/indexer/path/DatasourcePathSpec.java

```diff
@@ -0,0 +1,186 @@
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.indexer.path;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.metamx.common.logger.Logger;
+import io.druid.indexer.HadoopDruidIndexerConfig;
+import io.druid.indexer.hadoop.DatasourceIngestionSpec;
+import io.druid.indexer.hadoop.DatasourceInputFormat;
+import io.druid.indexing.overlord.IndexerMetadataStorageCoordinator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.timeline.DataSegment;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.MultipleInputs;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+public class DatasourcePathSpec implements PathSpec
+{
+  private static final Logger logger = new Logger(DatasourcePathSpec.class);
+
+  private IndexerMetadataStorageCoordinator indexerMetadataStorageCoordinator;
+  private ObjectMapper mapper;
+
+  private DatasourceIngestionSpec ingestionSpec;
+  private long maxSplitSize;
+
+  public DatasourcePathSpec(
+      @JacksonInject IndexerMetadataStorageCoordinator indexerMetadataStorageCoordinator,
+      @JacksonInject ObjectMapper mapper,
+      @JsonProperty("ingestionSpec") DatasourceIngestionSpec spec,
+      @JsonProperty("maxSplitSize") Long maxSplitSize
+  )
+  {
+    this.indexerMetadataStorageCoordinator = Preconditions.checkNotNull(indexerMetadataStorageCoordinator, "null indexerMetadataStorageCoordinator");
+    this.mapper = Preconditions.checkNotNull(mapper, "null mapper");
+    this.ingestionSpec = Preconditions.checkNotNull(spec, "null ingestionSpec");
+
+    if(maxSplitSize == null) {
+      this.maxSplitSize = 0;
+    } else {
+      this.maxSplitSize = maxSplitSize.longValue();
+    }
+  }
+
+  @JsonProperty
+  public DatasourceIngestionSpec getIngestionSpec()
+  {
+    return ingestionSpec;
+  }
+
+  @JsonProperty
+  public long getMaxSplitSize()
+  {
+    return maxSplitSize;
+  }
+
+  @Override
+  public Job addInputPaths(
+      HadoopDruidIndexerConfig config, Job job
+  ) throws IOException
+  {
+    final List<DataSegment> segments = indexerMetadataStorageCoordinator.getUsedSegmentsForInterval(
```

Comment:
hmmm, since SegmentListUsedAction and related infrastructure is present inside indexing-service which depends on indexing-hadoop, that means moving DatasourcePathSpec to indexing-service and somehow update jackson to understand new PathSpec without changing PathSpec interface.
I will need to check and see how easy/complicated it gets. Also, I dont think standalone hadoop index task will be able to work with that... should we deprecate standalone hadoop index task?